modules/hdf4_handler/HDF4RequestHandler.cc

```diff
@@ -327,7 +327,7 @@ cerr<<"Don't output ecs metadata "<<endl;
                 //throw InternalErr(__FILE__,__LINE__,"read_das_use_eos2lib error");
 #endif
             }
-            if(eosfile != NULL)
+            if(eosfile != nullptr)
```

Comment:
There's no pressing need to change this, but as you write new code, there is no need to test if the pointers are not null before calling 'delete' since the delete operator can safely be used on null pointers (unlike free).

Reply:
Thanks,  this will be handled later and I am going to remove the "new delete" with unique_pointer later. There are other cases that NULL needs to be changed to nullptr (like passing NULL as a parameter) if we want to conform to C11.  Also sonor cloud gives a lot of code smell. It buries other good warnings.

> here is no need to test if the pointers are not null before calling 'delete' since the delete operator can safely be used on null pointers (unlike free).